mqtt-protocol/build.gradle.kts

```diff
@@ -6,10 +8,167 @@ kotlin {
+            }
+        }
+
+        val tvosArm64Main by getting {
+            dependencies {
+                dependsOn(nativeMain)
+                implementation(files("native-libs/openssl-tvos-arm64.klib"))
+            }
+        }
+
+        val watchosSimulatorArm64Main by getting {
+            dependencies {
+                dependsOn(nativeMain)
+                implementation(files("native-libs/openssl-watchos-simulator-arm64.klib"))
+            }
+        }
+
+        val watchosArm64Main by getting {
+            dependencies {
+                dependsOn(nativeMain)
+                implementation(files("native-libs/openssl-watchos-arm64.klib"))
+            }
+        }
+
+        val iosX64Main by getting {
+            dependencies {
+                dependsOn(nativeMain)
+                implementation(files("native-libs/openssl-ios-x64.klib"))
             }
         }
+
+        val iosSimulatorArm64Main by getting {
+            dependencies {
+                dependsOn(nativeMain)
+                implementation(files("native-libs/openssl-ios-simulator-arm64.klib"))
+            }
+        }
+
+        val iosArm64Main by getting {
+            dependencies {
+                dependsOn(nativeMain)
+                implementation(files("native-libs/openssl-ios-arm64.klib"))
+            }
+        }
+
+        val macosArm64Main by getting {
+            dependencies {
+                dependsOn(nativeMain)
+                implementation(files("native-libs/openssl-macos-arm64.klib"))
+            }
+        }
+
+        val macosX64Main by getting {
+            dependencies {
+                dependsOn(nativeMain)
+                implementation(files("native-libs/openssl-macos-x64.klib"))
+            }
+        }
+
+        val mingwX64Main by getting {
+            dependencies {
+                dependsOn(nativeMain)
+                implementation(files("native-libs/openssl-mingw-x64.klib"))
+            }
+        }
+    }
+
+    val nativeSetup: KotlinNativeTarget.(targetName: String) -> Unit = { targetName ->
+        compilations["main"].defaultSourceSet.dependsOn(sourceSets[targetName+"Main"])
+    }
+
+    linuxX64 {
+        nativeSetup("linuxX64")
+    }
+
+    linuxArm64 {
+        nativeSetup("linuxArm64")
+    }
+
+    mingwX64 {
+        nativeSetup("mingwX64")
+    }
+
+    macosX64 {
+        nativeSetup("macosX64")
+    }
+
+    macosArm64 {
+        nativeSetup("macosArm64")
+    }
+
+    iosArm64 {
+        nativeSetup("iosArm64")
+    }
+
+    iosSimulatorArm64 {
+        nativeSetup("iosSimulatorArm64")
+    }
+
+    iosX64 {
+        nativeSetup("iosX64")
+    }
+
+    watchosArm64 {
+        nativeSetup("watchosArm64")
+    }
+
+    watchosSimulatorArm64 {
+        nativeSetup("watchosSimulatorArm64")
+    }
+
+    tvosArm64 {
+        nativeSetup("tvosArm64")
+    }
+
+    tvosSimulatorArm64 {
+        nativeSetup("tvosSimulatorArm64")
     }
```

Comment:
All of these lines should not be necessary since all the targets are enabled by default on the root level `build.gradle.kts`

Reply:
they are necessary for importing the correct openSSL file for KMQTT

Reply:
To import the openSSL files you only need the source sets, and they should be already available because of the configuration on the `build.gradle.kts`.
Have you tried removing the target declaration? If it does not work, what are the reported errors?

As a general note: don’t mark with resolved conversations that aren’t yet, I'll do it when I'm sure there is no room for further improvements :)

Reply:
If I remove those lines i get a runtime error 'Could not find "openssl" '

Reply:
Which lines have you tried to remove?

Reply:
i tried removing all the lines aside for the spicific sourcesets, for example for linuxArm64 when removing this lines, i get this error. Wich is why the CI on this PR was failing in the first place

val linuxArm64Main by getting {
  dependencies {
    dependsOn(nativeMain)
     implementation(files("native-libs/openssl-linux-arm64.klib"))
  }
}
linuxArm64 {
        nativeSetup("linuxArm64")
    }

Reply:
I meant to keep all the source sets with the openssl specification, but remove the target specification from lines 123-172.
Removing those lines should work as well